### PR TITLE
Recut - implement codespells, pre-commit, GitHub CI, double-check all files

### LINKED ISSUE #1068: make github workflow for building docs on merge to main

### LINKED ISSUE #1177: Add pip.txt in pyproject.toml
### Problem

`pip install -e .` does not install all dependencies under pip.txt

### Proposed solution

Add 

```python
[tool.setuptools.dynamic]
dependencies = {file = ["requirements/pip.txt"]}
```

Ref: https://github.com/diffpy/diffpy.utils/pull/124/files

### LINKED ISSUE #1179: Wondering purpose of `github_deploy_key.enc`
@sbillinge Could you please shed some light on the usage of this key? 

https://github.com/regro/regolith/blob/main/github_deploy_key.enc

### LINKED ISSUE #1180: Double check GitHub CI .yml files - there are more than enough of as now
### Problem

Screenshot 2024-11-06 at 4 11 56 PM

A bit too many.

### Proposed solution

Once pytest is configured, we can re-visit this and remove the ones no longer needed.

### PR DESCRIPTION
**Pull request by bobleesj**
Closes #1177 
Closes #1068 
Closes #1179 
Closes #1180 
Closes #1068 

### Local pytest result:

Python 3.11, `pip install -e .` shows the following:

Screenshot 2024-11-08 at 12 20 35 AM

### CODE REVIEW

File: doc/source/api/regolith.rst
Comment (bobleesj) [inline doc/source/api/regolith.rst:1]: These api references can be regenerated later

**Comment by bobleesj** [comment]
@sbillinge ready for review - as expected, test failing but only 3 fails, good news.

File: doc/source/conf.py [hunk 1]
```diff
@@ -76,7 +62,7 @@
 
 # General information about the project.
 project = "regolith"
-copyright = "2015-2017 Anthony Scopatz, 2018-%Y The Trustees of Columbia University in the City of New York"
+copyright = "%Y, The Trustees of Columbia University in the City of New York"
```
Comment (sbillinge) [inline doc/source/conf.py:65]: we don't want to delete Anthony copyright here.  Need to revert this change.

File: doc/source/conf.py [hunk 2]
```diff
@@ -104,7 +90,7 @@
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns = ["_build", "build"]
+exclude_patterns = ["build"]
```
Comment (sbillinge) [inline doc/source/conf.py:93]: please revert, regolith litters `_build`s around the place.

**Review by sbillinge — COMMENTED** [review]
it's likely that we want to revert most, if not all the changes in `doc/source/conf.py` and indeed it seems that this PR pretty much wipes out all the hand-written docs, so let's close it and go way more slowly.

**Comment by bobleesj** [comment]
Closing - re-cut again with a bit more careful approach, especially docs.
